audinterface/core/segment.py

```diff
@@ -555,9 +533,15 @@ def process_signal_from_index(
             task_description=f'Process {len(index)} segments',
         )
 
-        index = y[0]
-        for obj in y[1:]:
-            index = index.union(obj)
+        if len(index.levels) == 2:
+            # Index without 'file' level
+            index = y[0]
+            for obj in y[1:]:
+                index = index.union(obj)
+        else:
+            # For segmented index use audformat.utils.union()
+            # to ensure the correct dtypes are preserved
+            index = audformat.utils.union(y)
```

Comment:
If we would require `audformat>=0.15.3` we could reduce this further to just
```python
        index = audformat.utils.union(y)
```

Would there be any downsides in asking for a new `audformat` version?

Reply:
I don't think so, let's simplify.

Reply:
As the new audformat only works with Python >=3.8, I also had to remove support for Python 3.7.